osm.cc

```diff
@@ -789,9 +736,9 @@ osm_waypt_disp(const Waypoint* waypoint)
 
   waypoints.insert(name, waypoint);
 
-  int* id = (int*) xmalloc(sizeof(*id));
+  auto* id = new int;
   *id = --node_id;
-  (const_cast<Waypoint*>(waypoint))->extra_data = id;
+  const_cast<Waypoint*>(waypoint)->extra_data = id;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Assigning an integer to a pointer is not portable.](https://app.codacy.com/manual/gpsbabel/gpsbabel/pullRequest?prid=5617520)